plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/ProbabilisticSampler.java

```diff
@@ -37,27 +46,62 @@ public ProbabilisticSampler(TelemetrySettings telemetrySettings) {
         this.telemetrySettings = Objects.requireNonNull(telemetrySettings);
         this.samplingRatio = telemetrySettings.getSamplingProbability();
         this.defaultSampler = Sampler.traceIdRatioBased(samplingRatio);
+        this.actionBasedSampler = new HashMap<>();
+        this.actionBasedSamplingProbability = new HashMap<>();
     }
 
-    Sampler getSampler() {
+    /**
+     * Returns custom sampler based on the type of request
+     * @param attributes Telemetry attributes
+     */
+    Sampler getSampler(Attributes attributes) {
         double newSamplingRatio = telemetrySettings.getSamplingProbability();
```

Comment:
Please simplify this a bit in case action based sampling is defined why do you need to evaluate the SamplingRationChanged logic?

Reply:
Makes sense. Refactored to first check for action override, else will return default value.